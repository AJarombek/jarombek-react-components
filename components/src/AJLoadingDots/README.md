### Overview

Reusable loading component which displays three horizontal dots with a loading animation.

**Use cases:**

```javascript
import React from 'react';
import { AJLoadingDots } from 'jarombek-react-components';

const CustomComponent = () => 
  <AJLoadingDots className="custom-class" />
```

### Files

| Filename                   | Description                                                                           |
|----------------------------|---------------------------------------------------------------------------------------|
| `AJLoadingDots.js`         | The reusable `AJLoadingDots` component.                                               |
| `index.js`                 | Index file which exports the `AJLoadingDots` component.                               |
| `styles.js`                | JSS styles for the `AJLoadingDots` component.                                         |
