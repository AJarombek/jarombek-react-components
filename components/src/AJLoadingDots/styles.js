export default {
  container: {
    display: 'grid',
    gridTemplateColumns: 'repeat(3, 1fr)',
    gridTemplateRows: '100%',
  },
  circle: {
    backgroundColor: '#333',
    width: 40,
    height: 40,
    borderRadius: '50%',
    margin: 5,
    opacity: 0,
    animation: '$fade 1s infinite',
  },
  first: {
    animationDelay: '0s',
  },
  second: {
    animationDelay: '0.25s',
  },
  third: {
    animationDelay: '0.5s',
  },
  '@keyframes fade': {
    '0%': { opacity: 0 },
    '20%': { opacity: 1 },
    '100%': { opacity: 0 },
  },
};
