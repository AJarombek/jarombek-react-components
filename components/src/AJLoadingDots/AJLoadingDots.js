/**
 * Reusable loading component which displays three horizontal dots.
 * @author Andrew Jarombek
 * @since 5/20/2018
 */

import React from 'react';
import PropTypes from "prop-types";
import {createUseStyles} from 'react-jss';
import classnames from 'classnames';
import styles from './styles';

const useStyles = createUseStyles(styles);

const AJLoadingDots = ({className}) => {
  const classes = useStyles();

  return (
    <div className={classnames('aj-loading-dots', classes.container,  className)}>
      <div className={classnames('jarbek-loading-circle', classes.circle, classes.first)} />
      <div className={classnames('jarbek-loading-circle', classes.circle, classes.second)} />
      <div className={classnames('jarbek-loading-circle', classes.circle, classes.third)} />
    </div>
  );
};

AJLoadingDots.propTypes = {
  className: PropTypes.string
};

export default AJLoadingDots;
