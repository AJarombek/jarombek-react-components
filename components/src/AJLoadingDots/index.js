import AJLoadingDots from './AJLoadingDots';

export default AJLoadingDots;
