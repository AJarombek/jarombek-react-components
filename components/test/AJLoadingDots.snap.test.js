/**
 * Snapshot test for the {@link ../../components/AJLoadingDots} component.
 * @author Andrew Jarombek
 * @since 8/14/2020
 */

import React from 'react';
import AJLoadingDots from '../src/AJLoadingDots/AJLoadingDots';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<AJLoadingDots/>).toJSON();
  expect(tree).toMatchSnapshot();
});
