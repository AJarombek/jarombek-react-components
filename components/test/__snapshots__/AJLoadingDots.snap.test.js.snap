// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="aj-loading-dots container-0-2-1"
>
  <div
    className="jarbek-loading-circle circle-0-2-2 first-0-2-3"
  />
  <div
    className="jarbek-loading-circle circle-0-2-2 second-0-2-4"
  />
  <div
    className="jarbek-loading-circle circle-0-2-2 third-0-2-5"
  />
</div>
`;
